electron/ironfish/AccountManager.ts

```diff
@@ -83,18 +93,45 @@ class AccountManager implements IIronfishAccountManager {
 
   async balance(
     id: string,
-    assetId: Buffer = Asset.nativeId()
+    assetId: string = NativeAsset.nativeId().toString('hex')
   ): Promise<AccountBalance> {
     const account = this.node.wallet.getAccount(id)
     if (account) {
-      const balance = await this.node.wallet.getBalance(account, assetId)
-      const asset = await this.node.chain.getAssetById(assetId)
+      const balance = await this.node.wallet.getBalance(
+        account,
+        Buffer.from(assetId, 'hex')
+      )
+      const asset = await this.assetManager.get(assetId)
       return {
         ...balance,
-        asset: {
-          id: asset.id.toString('hex'),
-          name: asset?.name.toString('utf8') || '',
-        },
+        asset: asset,
+      }
+    }
+
+    return Promise.reject(new Error(`Account with id=${id} was not found.`))
+  }
+
+  async balances(id: string): Promise<{
+    default: AccountBalance
+    assets: AccountBalance[]
+  }> {
+    const account = this.node.wallet.getAccount(id)
+    if (account) {
+      const balances: AccountBalance[] = []
+      for await (const balance of this.node.wallet.getBalances(account)) {
+        const asset: Asset = await this.assetManager.get(balance.assetId)
+
+        balances.push({
+          ...balance,
+          asset: asset,
+        })
+      }
+
+      const defaultAssetID = NativeAsset.nativeId().toString('hex')
+
+      return {
+        default: balances.find(b => b.asset.id === defaultAssetID),
+        assets: balances.filter(b => b.asset.id !== defaultAssetID),
```

Comment:
suggestion: Totally up to you, but IMO guard clauses read a little better since you have to nest less of the code.

```tsx
  async balances(id: string): Promise<{
    default: AccountBalance
    assets: AccountBalance[]
  }> {
    const account = this.node.wallet.getAccount(id)

    if (!account) {
      throw new Error(`Account with id=${id} was not found.`)
    }

    const balances: AccountBalance[] = []
    for await (const balance of this.node.wallet.getBalances(account)) {
      const asset: Asset = await this.assetManager.get(balance.assetId)

      balances.push({
        ...balance,
        asset: asset,
      })
    }

    const defaultAssetID = NativeAsset.nativeId().toString('hex')

    return {
      default: balances.find(b => b.asset.id === defaultAssetID),
      assets: balances.filter(b => b.asset.id !== defaultAssetID),
    }
  }
```

Reply:
Refactored